Tests/SwiftlyTests/SwiftlyTests.swift

```diff
@@ -6,11 +6,71 @@ import NIO
 @testable import SwiftlyCore
 import XCTest
 
+#if os(macOS)
+import MacOSPlatform
+#endif
+
+import AsyncHTTPClient
+import NIO
+
 struct SwiftlyTestError: LocalizedError {
     let message: String
 }
 
+/// An `HTTPRequestExecutor` backed by an `HTTPClient` that can take http proxy
+/// information from the environment in either HTTP_PROXY or HTTPS_PROXY
+class ProxyHTTPRequestExecutorImpl: HTTPRequestExecutor {
+    let httpClient: HTTPClient
+    public init() {
+        var proxy: HTTPClient.Configuration.Proxy?
+
+        let environment = ProcessInfo.processInfo.environment
+        let httpProxy = environment["HTTP_PROXY"]
+        if let httpProxy, let url = URL(string: httpProxy), let host = url.host, let port = url.port {
+            proxy = .server(host: host, port: port)
+        }
+
+        let httpsProxy = environment["HTTPS_PROXY"]
+        if let httpsProxy, let url = URL(string: httpsProxy), let host = url.host, let port = url.port {
+            proxy = .server(host: host, port: port)
+        }
+
+        if proxy != nil {
+            self.httpClient = HTTPClient(eventLoopGroupProvider: .singleton, configuration: HTTPClient.Configuration(proxy: proxy))
+        } else {
+            self.httpClient = HTTPClient.shared
+        }
+    }
+
+    public func execute(_ request: HTTPClientRequest, timeout: TimeAmount) async throws -> HTTPClientResponse {
+        try await self.httpClient.execute(request, timeout: timeout)
+    }
+
+    deinit {
+        if httpClient !== HTTPClient.shared {
+            try? httpClient.syncShutdown()
+        }
+    }
+}
+
 class SwiftlyTests: XCTestCase {
+    private static var requestExecutor: ProxyHTTPRequestExecutorImpl?
+
+    override class func setUp() {
+        if Self.requestExecutor == nil {
+            Self.requestExecutor = ProxyHTTPRequestExecutorImpl()
+            Install.httpClient = SwiftlyHTTPClient(executor: Self.requestExecutor)
+            SelfUpdate.httpClient = SwiftlyHTTPClient(executor: Self.requestExecutor)
+        }
+    }
```

Comment:
I think we'll need to unconditionally reset all the httpClients to use the proxy executor, since many tests will replace them with mocked http clients. To make this easier, it would probably be preferable to have one single global singleton HTTP client defined in SwiftlyCore or something.